core/src/main/java/io/questdb/griffin/engine/functions/table/TouchTableFunctionFactory.java

```diff
@@ -0,0 +1,174 @@
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2020 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.table;
+
+import io.questdb.TelemetryJob;
+import io.questdb.cairo.BitmapIndexReader;
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.*;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.StrFunction;
+import io.questdb.griffin.engine.functions.UnaryFunction;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.std.*;
+import io.questdb.std.str.CharSink;
+import io.questdb.std.str.StringSink;
+
+public class TouchTableFunctionFactory implements FunctionFactory {
+    @Override
+    public String getSignature() {
+        return "touch(C)";
+    }
+
+    @Override
+    public Function newInstance(int position,
+                                ObjList<Function> args,
+                                IntList argPositions,
+                                CairoConfiguration configuration,
+                                SqlExecutionContext sqlExecutionContext
+    ) throws SqlException {
+        final Function function = args.get(0);
+        final int pos = argPositions.get(0);
+
+        try(final RecordCursorFactory recordCursorFactory = function.getRecordCursorFactory()) {
+            if (recordCursorFactory == null || !recordCursorFactory.supportPageFrameCursor()) {
+                throw SqlException.$(pos, "query does not support framing execution and cannot be pre-touched");
+            }
+        }
+
+        return new TouchTableFunc(function);
+    }
+
+    private static class TouchTableFunc extends StrFunction implements UnaryFunction {
+        private static final Log LOG = LogFactory.getLog(TouchTableFunc.class);
+
+        private final Function arg;
+
+        private SqlExecutionContext sqlExecutionContext;
+
+        private final StringSink sinkA = new StringSink();
+        private final StringSink sinkB = new StringSink();
+
+        private long garbage = 0;
```

Comment:
`garbage` is not used, could go

Reply:
or could be also reported by getStr

Reply:
garbage used to block compiler optimization